### DISCUSSION TITLE
Help with IntegerNumbers

### DISCUSSION THREAD

**Discussion by BIMster008** [buildingSMART/IDS#368] [Q&A]
Hi,
after I got help with the predifinedType, I managed to write almost 5000 lines of code. Sadly got stuck in a CustomSet.
I have numerical (integer) values in a customset which carry the datatype IFCLENGTHMEASURE.
Here is the line in editor:

```
#2888323=IFCPROPERTYSINGLEVALUE('Laenge',$,IFCLENGTHMEASURE(7.250008008868917),$);
#2888324=IFCPROPERTYSINGLEVALUE('Hoehe',$,IFCLENGTHMEASURE(3.419999999999998),$);
```

Therefore I wrote also the xml lines with the Set, property and Datatype in this way:

```
                <ids:property dataType="IFCLENGTHMEASURE">
                    <ids:propertySet>
                        <ids:simpleValue>AUKOset</ids:simpleValue>
                    </ids:propertySet>
                    <ids:baseName>
                        <ids:simpleValue>Laenge</ids:simpleValue>
                    </ids:baseName>
                </ids:property>
                <ids:property dataType="IFCLENGTHMEASURE">
                    <ids:propertySet>
                        <ids:simpleValue>AUKOset</ids:simpleValue>
                    </ids:propertySet>
                    <ids:baseName>
                        <ids:simpleValue>Hoehe</ids:simpleValue>
                    </ids:baseName>
                </ids:property>
```

All valuew are present in IFC in different forms such as 0.5 or 124.417
I have tried diferent datatypes: IFCINTEGER, IFCPOSITIVEINTEGER and even IFCLABEL. None worked.
I tried even renaming the property to quantity. Didn't work
I am running out of ideas... Bonsai is marking them red :(

The only thing coming to my mind is the way such numbers are written. In Austria, we write decimals number  like 0,5 instead of 0.5
Might this be maybe a problem?
Thank you

**Comment by atomczak** [comment]
Is Bonsai marking the IDS file red (its correctness), or the IFC in the process of IDS checking? You can check the first here: https://www.xbim.it/ids
Also, it's hard to tell without full ifc and ids files. Can you share those two as zip here?

**Comment by BIMster008** [comment]
hi @atomczak 
I must admit, that IDS has it's working and not working days. I ran then same code as yesterday with IFCLENGTHMEASURE and it worked. I almost fell out of the chair :)
I still got the red flags in HTML, but this time about faulty elements and not a message such as "The property's data type "IfcLabel" does not match the required data type of "IFCLENGTHMEASURE". That's the result that I wanted to achieve.
Everything seems to be working fine, and I cannot explain the faulty results of the past days....

Thank you for the xbim hint. The tool seems pretty cool. I ran my code there and got quite a few errors. This is the one with the most repetitions:
```
Error Error 303: Invalid base type; empty string on `restriction` element at line 107, position 26.

Error Error 303: Invalid base type for Ifc4; found empty but expected `xs:string` for `IFCLABEL` on `value` element at line 106, position 22.
```

The result is talking about the lines (106) <ids:value> and (107)  <xs:restriction> 
```
                <ids:property dataType="IFCLABEL">
                    <ids:propertySet>
                        <ids:simpleValue>TVG_Spezifisch</ids:simpleValue>
                    </ids:propertySet>
                    <ids:baseName>
                        <ids:simpleValue>Sichtqualitaet</ids:simpleValue>
                    </ids:baseName>
                    <ids:value>
                        <xs:restriction>
                            <xs:enumeration value="NE" />
                            <xs:enumeration value="SB0" />
                            <xs:enumeration value="SB1" />
                            <xs:enumeration value="SB2" />
                            <xs:enumeration value="SB3" />
                            <xs:enumeration value="SB4" />
                        </xs:restriction>
                    </ids:value>
                </ids:property>
```
Frankly speaking, I have no idea what the error means.
Other errors pointed to a mistake that I had overseen such as the non-existence of property "LoadBearing" in IfcFooting in IFC4. I love it 😄

**Reply by atomczak** [reply]
you miss the base: <xs:restriction base="xs:string">. I recommend using existing software implementations that will help you build IDSes. Writting the XML code manually is error-prone.

**Reply by BIMster008** [reply]
@atomczak 
I corrected this error and it looks better. 
I am interested of what is going on under the hood 🧠 
I really appreciate your patience to explain every step, and I am very thankful.
Now I am on the path to success.
thanx
